### ISSUE TITLE
[$250] Search-Invoice with +2 expense submitted in OD is displayed with a negative total amount in ND

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53045]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.66-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5256232
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Login with a account that is an admin of a workspace
2. Login with the same account in OD
3. In OD - Submit a invoice to anyone with at least 2 expenses
4. Back in ND - Navigate to the Search page
5. Click on invoices
6. Click on the Outstanding filter
7. Verify the expenses are now divided per invoice report
8. Verify the invoice submitted in OD has the 2 expenses grouped

## Expected Result:
The invoice with at least 2 expenses submitted in OD is displayed with a positive total amount in ND.

## Actual Result:
The invoice with at least 2 expenses submitted in OD is displayed with a negative total amount in ND.

## Workaround:
Unknown

## Platforms:

- [x] Android: Standalone
- [x] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
![Bug6674480_1732394559440!total_negative_amount](https://github.com/user-attachments/assets/95cbfc6d-f678-438c-8df9-dec4dccc4906)

https://github.com/user-attachments/assets/65bb2063-5079-4993-baf8-85d3fec1d880

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021861216736472283553</li>
        <li>Upwork Job ID: 1861216736472283553</li>
        <li>Last Price Increase: 2024-12-17</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @ikevin127</details>

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The total invoice report amount is shown in negative.

### What is the root cause of that problem?
The total amount of the invoice report is in negative, the same as expense report. However, when we show the total amount, we only negate the expense report.
https://github.com/Expensify/App/blob/f4f8da74c100f2c5ddb666ff4314038010104cbf/src/components/SelectionList/Search/ReportListItem.tsx#L39-L44

So, for invoice report, the negative amount is shown.

### What changes do you think we should make in order to solve the problem?

We need to check for invoice type too and if it's true, then we negate it too (multiply by -1)
```
total *= (reportItem?.type === CONST.REPORT.TYPE.EXPENSE || reportItem?.type === CONST.REPORT.TYPE.INVOICE) ? -1 : 1;

OR

total *= (isExpenseReport(reportItem) || isInvoiceReport(reportItem)) ? -1 : 1;
```
(or we can use Math.abs if it's not an expense report)

There are multiple places where we check for expense report type to multiply it by -1 where we can probably apply the same fix, but looks like it's currently not possible to view an invoice report, so not sure how to test the other cases.

**Comment by mallenexpensify** [comment]
> There are multiple places where we check for expense report type to multiply it by -1 where we can probably apply the same fix,

Music to my ears @bernhardoj :)

Was able to reproduce
<img width="979" alt="SnagitHelper2024 2024-11-25 17 13 16" src="https://github.com/user-attachments/assets/baa49949-f5ec-464e-95ea-53768ec11894">

When I click on `View` I get an error, is that also related here? 
<img width="973" alt="SnagitHelper2024 2024-11-25 17 13 32" src="https://github.com/user-attachments/assets/3e267d59-f438-4e67-9a7f-719801027a37">

<img width="978" alt="Google Chrome 2024-11-25 17 13 24" src="https://github.com/user-attachments/assets/ef50b462-dc4f-4a34-b44f-e882fa57cdee">

Only error I see in console
<img width="369" alt="image" src="https://github.com/user-attachments/assets/e67c1582-ffb6-4f11-86a0-e904a6992953">

**Comment by bernhardoj** [comment]
The BE returns not found error when viewing it.
<img width="518" alt="image" src="https://github.com/user-attachments/assets/e1134a36-f993-486f-99d7-4252c3077538">

**Comment by ikevin127** [comment]
Let's go with @bernhardoj's [proposal](https://github.com/Expensify/App/issues/53045#issuecomment-2496687504) since the RCA makes sense and the proposed solution fixes the issue.

Regarding the solution, since the component already imports `ReportUtils`, I like this version as it's 14 characters shorter:

```ts
total *= ReportUtils.isExpenseReport(reportItem) || ReportUtils.isInvoiceReport(reportItem) ? -1 : 1;
```

🎀👀🎀 C+ reviewed

##

> When I click on `View` I get an error, is that also related here ?

I don't see it related to the negative amount issue in any way which would justify extending the scope of this issue given the context shared in https://github.com/Expensify/App/issues/53045#issuecomment-2499391353 - I think it should be treated as a separate issue.

**Comment by bernhardoj** [comment]
Updated my proposal to include the util function. 

> as it seems that code is also wrong

Which part of `TransactionUtils.getAmount` is wrong?

**Comment by iwiznia** [comment]
Actually not sure if it is wrong, because I now realize I have no idea what `isFromTrackedExpense` means (let's add a comment on the param as part of this change once we figure it out). I was thinking it is not checking invoices there, so it might be using the abs version instead of the negated version.

**Comment by bernhardoj** [comment]
My first thought that it should be true for self DM track, but I just realized we can track expense in workspace too. It was added in https://github.com/Expensify/App/pull/38709, specifically this [commit](https://github.com/Expensify/App/pull/38709/commits/bb4483fb5877b1121137e650c603896212d47844), but looks like we never make use of the `isFromTrackedExpense` param in that PR, meaning it's always false.

The only code that uses it is this:
https://github.com/Expensify/App/blob/64eaf2fdd357bd06008adfb9c0c049e09cd375b9/src/pages/iou/request/step/IOURequestStepTaxRatePage.tsx#L65

It's from this PR: https://github.com/Expensify/App/pull/40443

Maybe @ishpaul777 can help us clarify the purpose of that param.

**Comment by iwiznia** [comment]
ccing @thienlnam too since they were reviewer of https://github.com/Expensify/App/pull/38709

**Comment by ikevin127** [comment]
> Waiting on input from the people pinged above

**Comment by ikevin127** [comment]
Still working on a decision regarding the direction of this issue and its fix.

**Comment by mallenexpensify** [comment]
@thienlnam 👀 above plz cuz you were the reviewier on the PR below. 
> we can track expense in workspace too. It was added in https://github.com/Expensify/App/pull/38709,

**Comment by thienlnam** [comment]
Could someone summarize what the confusion is regarding?

If it's about the isFromTrackedExpense param, it's been a while but my guess is from when you convert a tracked expense to a reimbursable one since we flip the amount (negative to positive). If it's not used anymore, let's just remove it

**Comment by bernhardoj** [comment]
Yes, it's the `isFromTrackedExpense` param, but we have one usage of it on the IOU tax rate page.

https://github.com/Expensify/App/blob/64eaf2fdd357bd06008adfb9c0c049e09cd375b9/src/pages/iou/request/step/IOURequestStepTaxRatePage.tsx#L65

From this [commit](https://github.com/Expensify/App/commit/228c47705cc4a06d8d0a343be222e8e952523039), previously we get the transaction amount by `transaction.amount` to calculate the tax amount. That PR updates it to use `TransactionUtils.getAmount` most likely to cover the case when the amount is updated, thus we should get from `transaction.modifiedAmount` instead (which what `TransactionUtils.getAmount` does).

My guess with why we pass `isFromTrackedExpense` as true and `isFromExpenseReport` as false is so that the amount returned is always in positive.

https://github.com/Expensify/App/blob/3baf965618ced48845ddd7534b0fad4d4110b5de/src/libs/TransactionUtils/index.ts#L423-L430

I think we can safely remove `isFromTrackedExpense`. @iwiznia What do you think?

**Comment by ikevin127** [comment]
Not overdue - still discussing.

**Comment by ikevin127** [comment]
Not overdue - still discussing.

**Comment by ikevin127** [comment]
Awaiting feedback from @iwiznia regarding https://github.com/Expensify/App/issues/53045#issuecomment-2522179662 so we can decide whether we can move forward with the selected proposal or should the solution be adjusted.

**Comment by iwiznia** [comment]
I have no idea if we can safely remove it because I don't get what it does and don't know where it is used. Did you check the usages to ensure we don't need it?

**Comment by bernhardoj** [comment]
I believe yes since we only have 1 usage as mentioned in my previous [comment](https://github.com/Expensify/App/issues/53045#issuecomment-2522179662). Also, if we remove the `isFromTrackedExpense`, it won't change anything because `isFromExpenseReport` is passed as false (as shown in my previous comment too).

**Comment by iwiznia** [comment]
Ah ok, I was confused because I saw more usages, but now I see you meant no usages passing `true`. I agree on removing it then

**Comment by bernhardoj** [comment]
Okay great, waiting for an assignment then.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by ikevin127** [comment]
> Issue not reproducible during KI retests. (First week)

@bernhardoj Can you confirm whether the issue is still reproducible on your side ? If it is, feel free to tag **[iwiznia](https://github.com/iwiznia)** for assignment based on my review from https://github.com/Expensify/App/issues/53045#issuecomment-2499627849.

**Comment by bernhardoj** [comment]
I can't repro anymore. It's been fixed by https://github.com/Expensify/App/pull/52556

**Comment by mallenexpensify** [comment]
Thanks @ikevin127 and @bernhardoj , closing as it's not reproducible
